### PR TITLE
feat(auth): Add ECS metadata API authentication support

### LINKED ISSUE #1747: [Authentication] Add new ecs agency metadata method support
I'd like to add a new authentication method to leverage ecs agency metadata securitykey.

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1747 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [ ] Tests added/passed.
* [x] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccSomethingV0_basic'
...
=== RUN   TestAccSomethingV0_basic
--- PASS: TestAccSomethingV0_basic (70.75s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       70.796s
```

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v2
       with:
-        go-version: 1.15
+        go-version: 1.16
```
Comment (ShiChangkuo) [inline .github/workflows/ci.yml]: suggest to update the go version in a new PR

File: huaweicloud/config/config.go [hunk 2]
```diff
@@ -371,6 +401,59 @@ func genClients(c *Config, pao, dao golangsdk.AuthOptionsProvider) error {
 	return err
 }
 
+func getAuthConfigByMeta(c *Config) error {
+	req, err := http.NewRequest("GET", securityKeyURL, nil)
+	req.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return fmt.Errorf("Error building metadata API request: %s", err.Error())
+	}
+	httpClient := &http.Client{}
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go:407]: should we add the user-agent and log the request?

File: .github/workflows/cron-ci.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v2
       with:
-        go-version: 1.15
+        go-version: 1.16
```
Comment (ShiChangkuo) [inline .github/workflows/cron-ci.yml:23]: suggest to update the go version in a new PR

File: huaweicloud/config/config.go [hunk 4]
```diff
@@ -89,7 +97,14 @@ func (c *Config) LoadAndValidate() error {
...
 			err = buildClientByPassword(c)
 		}
-
+	} else {
+		err = getAuthConfigByMeta(c)
+		if err != nil {
+			return fmt.Errorf("Error fetching Auth credentials from ECS Metadata API, AkSk or ECS agency must be provided: %s", err)
+		}
+		expairesAt, _ := time.Parse(time.RFC3339, c.SecurityKeyExpiresAt)
+		log.Printf("Successfully got metadata security key, which will expire at: %s", expairesAt)
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go]: log-level is missing

File: huaweicloud/config/config.go [hunk 5]
```diff
@@ -426,6 +518,15 @@ func (c *Config) NewServiceClient(srv, region string) (*golangsdk.ServiceClient,
 		return nil, fmt.Errorf("service type %s is invalid or not supportted", srv)
 	}
 
+	if c.SecurityKeyExpiresAt != "" {
+		timeNow := time.Now().Unix()
+		expairesAt, _ := time.Parse(time.RFC3339, c.SecurityKeyExpiresAt)
+		expairesAtInt := expairesAt.Unix()
+		if timeNow+keyExpiresDuration > expairesAtInt {
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go:528]: suggest to ues `time.Since(xxx).Seconds() > expairesAtInt`

File: huaweicloud/config/config.go [hunk 6]
```diff
@@ -89,7 +97,14 @@ func (c *Config) LoadAndValidate() error {
 		} else {
 			err = buildClientByPassword(c)
 		}
-
+	} else {
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go:104]: should we add a switch in provider?

File: huaweicloud/config/config.go [hunk 7]
```diff
@@ -410,6 +493,15 @@ func (c *Config) ObjectStorageClient(region string) (*obs.ObsClient, error) {
 		}
 	}
 
+	if c.SecurityKeyExpiresAt != "" {
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go]: suggest to add a lock

File: huaweicloud/config/config.go [hunk 8]
```diff
@@ -55,6 +60,9 @@ type Config struct {
 	RegionClient        bool
 	EnterpriseProjectID string
 
+	// metadata security key expires at
+	SecurityKeyExpiresAt string
```
Comment (ShiChangkuo) [inline huaweicloud/config/config.go]: suggest to change the type to time.Time

Comment (niuzhenguo) [reply huaweicloud/config/config.go]: done

File: huaweicloud/config/config.go (hunk 7 above)
Comment (niuzhenguo) [reply huaweicloud/config/config.go]: done

File: huaweicloud/config/config.go (hunk 6 above)
Comment (niuzhenguo) [reply huaweicloud/config/config.go:104]: I'd like to refactor this part later on with unionsdk.

File: huaweicloud/config/config.go (hunk 5 above)
Comment (niuzhenguo) [reply huaweicloud/config/config.go:528]: as we discussed, keep it as is

File: huaweicloud/config/config.go (hunk 4 above)
Comment (niuzhenguo) [reply huaweicloud/config/config.go]: done

File: huaweicloud/config/config.go (hunk 2 above)
Comment (niuzhenguo) [reply huaweicloud/config/config.go:407]: done

File: .github/workflows/ci.yml (hunk 1 above)
Comment (niuzhenguo) [reply .github/workflows/ci.yml]: done

**Pull request merged**